src/app/client/src/app/modules/public/module/explore/components/explore-curriculum-courses/explore-curriculum-courses.component.ts

```diff
@@ -0,0 +1,135 @@
+  templateUrl: './explore-curriculum-courses.component.html',
+  styleUrls: ['./explore-curriculum-courses.component.scss']
+})
+export class ExploreCurriculumCoursesComponent implements OnInit, OnDestroy {
+  public channelId: string;
+  public isCustodianOrg = true;
+  private unsubscribe$ = new Subject<void>();
+  public defaultBg = false;
+  public defaultFilters = {
+    board: [DEFAULT_FRAMEWORK],
+    gradeLevel: ['Class 10'],
+    medium: []
+  };
+  private subjectThemeAndIconsMap = {
+    Science: {
+      background: '#FFD6EB',
+      titleColor: '#FD59B3',
+      icon: './../../../../../../../assets/images/science.svg'
+    },
+    Mathematics: {
+      background: '#FFDFD9',
+      titleColor: '#EA2E52',
+      icon: './../../../../../../../assets/images/mathematics.svg'
+    },
+    English: {
+      background: '#DAFFD8',
+      titleColor: '#218432'
+    },
+    Social: {
+      background: '#DAD4FF',
+      titleColor: '#635CDC',
+      icon: './../../../../../../../assets/images/social.svg'
+    }
+  };
+
+  public selectedCourse;
+  public courseList: Array<{}> = [];
+  public title: string;
+
+  constructor(private searchService: SearchService, private toasterService: ToasterService, private userService: UserService,
+    public resourceService: ResourceService, private configService: ConfigService, public activatedRoute: ActivatedRoute,
+    private router: Router, private orgDetailsService: OrgDetailsService, private navigationhelperService: NavigationHelperService,
+    private contentSearchService: ContentSearchService) { }
+
+    ngOnInit() {
+      this.title = _.get(this.activatedRoute, 'snapshot.queryParams.title');
+      this.defaultFilters = _.omit(_.get(this.activatedRoute, 'snapshot.queryParams'), 'title');
+        this.getChannelId().pipe(
+          mergeMap(({ channelId, isCustodianOrg }) => {
+            this.channelId = channelId;
+            this.isCustodianOrg = isCustodianOrg;
+            return this.contentSearchService.initialize(channelId, isCustodianOrg, this.defaultFilters.board[0]);
+          }),
+          takeUntil(this.unsubscribe$))
+          .subscribe(() => {
+            this.fetchCourses();
+          }, (error) => {
+            this.toasterService.error(this.resourceService.frmelmnts.lbl.fetchingContentFailed);
+            setTimeout(() => this.router.navigate(['']), 5000);
+            console.error('init search filter failed', error);
+          });
+    }
+
+    private getChannelId() {
+      if (this.userService.slug) {
+        return this.orgDetailsService.getOrgDetails(this.userService.slug)
+          .pipe(map(((orgDetails: any) => ({ channelId: orgDetails.hashTagId, isCustodianOrg: false }))));
+      } else {
+        return this.orgDetailsService.getCustodianOrg()
+          .pipe(map(((custOrgDetails: any) => ({ channelId: _.get(custOrgDetails, 'result.response.value'), isCustodianOrg: true }))));
+      }
+    }
+
+    private getSearchRequest() {
```

Comment:
Move getSearchRequest and fetchCourses to service and component should call one method from service to get the courses in the required format as observable or should throw error